src/dso_hdr.cc

```diff
@@ -112,7 +114,12 @@ uint64_t DsoStats::sum_event_metric(DsoEventType dso_event) const {
 DsoHdr::DsoHdr() {
   // keep dso_id 0 as a reserved value
   // Test different places for existence of /proc
-  if (check_file_type("/host/proc", S_IFDIR)) {
+  // A given procfs can only work if its PID namespace is the same as mine.
+  // Fortunately, `/proc/self` will return a symlink to my process ID in the
+  // corresponding namespace, so this is easy to check
+  char pid_str[sizeof("42949672960")] = {};
```

Comment:
[minor] I thought max pid was `4194304`. Regardless this is fine, I just wanted to understand the logic

Reply:
My idea was that the kernel uses the entire `pid_t`, so I thought of using `2^32`, but now that I review https://github.com/torvalds/linux/blob/master/include/linux/threads.h#L30 I see that the actual _numerical_ limit is `2^30` (nothing larger is ever actually used as a PID and will never appear in traces or in userspace).  I'll change it and add a clarifying comment, since otherwise the number is mysterious.  Thanks!

Reply:
Fixed in latest commit